applications/crossbar/src/cb_pager.erl

```diff
@@ -0,0 +1,54 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2011-2016, 2600Hz
+%%% @doc
+%%%
+%%% @end
+%%% @contributors
+%%%   Roman Galeev
+%%%-------------------------------------------------------------------
+-module(cb_pager).
+-export([descending/7, ascending/7]).
+
+-spec descending(binary(), binary(), integer(), integer(), integer(), fun(), []) -> {integer(), list()}.
+descending(AccountId, View, Start, End, Limit, Filter, Options) when
+      is_binary(AccountId), is_binary(View), is_integer(Start), is_integer(End), is_integer(Limit), Start > End ->
+    MODbs = lists:reverse(kazoo_modb:get_range(AccountId, End, Start)),
+    CouchOpts = [{'startkey', Start}, {'endkey', End}, 'descending' | Options],
+    {_, LastKey, JObjs} =
+        lists:foldl(fun fold_query/2, {Limit, 'undefined', []}, [ {Db, View, CouchOpts, Filter} || Db <- MODbs ]),
+    {LastKey, JObjs}.
+
+-spec ascending(binary(), binary(), integer(), integer(), integer(), fun(), []) -> {integer(), list()}.
+ascending(AccountId, View, Start, End, Limit, Filter, Options) when
+      is_binary(AccountId), is_binary(View), is_integer(Start), is_integer(End), is_integer(Limit), Start < End ->
+    MODbs = kazoo_modb:get_range(AccountId, Start, End),
+    CouchOpts = [{'startkey', Start}, {'endkey', End} | Options],
+    {_, LastKey, JObjs} =
+        lists:foldl(fun fold_query/2, {Limit, 'undefined', []}, [ {Db, View, CouchOpts, Filter} || Db <- MODbs ]),
+    {LastKey, JObjs}.
+
+-spec fold_query({binary(), binary(), list(), fun()}, {integer(), integer(), list()}) -> {integer(), integer(), list()}.
+fold_query({Db, View, CouchOpts, Filter}, {Limit, _LastKey, Res}) when is_integer(Limit), Limit > 0 ->
+    Queried = erlang:length(Res),
+    LimitWithLast = 1 + Limit - Queried,
+    DbResults = limited_query(LimitWithLast, Db, View, CouchOpts),
+    {LastKey, JObjs} = last_key(lists:reverse(DbResults), LimitWithLast, erlang:length(DbResults)),
```

Comment:
Could we get better time complexity? `reverse` + `length` + more `reverse` in last_key's code